### ISSUE TITLE
Crashes when starting an application using ES6 modules

### ISSUE THREAD

**Issue by jpike88** [microsoft/vscode-node-debug#165]
Attached verbose dump:

[vscode-node-debug2.txt](https://github.com/Microsoft/vscode-node-debug/files/1500068/vscode-node-debug2.txt)

**Comment by jpike88** [comment]
When using integrated terminal, I get this error:

```
Debugger listening on ws://127.0.0.1:4363/ffc5c2f7-0765-4396-b0d5-6548fc3e238a
For help see https://nodejs.org/en/docs/inspector
Debugger attached.
(node:65673) ExperimentalWarning: The ESM module loader is experimental.
Error: linking error, dependency promises must be resolved on instantiate
    at checkComplete (internal/loader/ModuleJob.js:75:27)
    at moduleJob.linked.then (internal/loader/ModuleJob.js:58:11)
    at <anonymous>
Waiting for the debugger to disconnect...
```

**Comment by jpike88** [comment]
Just tried with Chrome debugger directly, same error... looks like VSCode may not be the issue here

![screen shot 2017-11-24 at 2 06 50 am](https://user-images.githubusercontent.com/9585787/33186438-3487e470-d0bc-11e7-86a7-13614ebbd863.png)

**Comment by jpike88** [comment]
Closing as this is not VSCode's problem

**Comment by jpike88** [comment]
I've opened an issue here:

https://github.com/nodejs/node/issues/17280

**Comment by jpike88** [comment]
Actually, this is still relevant, I'll reopen unless someone decides otherwise

**Comment by jpike88** [comment]
The problem is that the debugger attaches too early. It needs breathing space.

**Comment by roblourens** [comment]
Delay until what?

**Comment by jpike88** [comment]
Perhaps give it 2 or 3 seconds, as it's currently interfering with ES6 module linking.
See:

https://github.com/nodejs/node/issues/17280

**Comment by roblourens** [comment]
It's node's bug, but you can use one of the workarounds from the other issue.

**Comment by jpike88** [comment]
So my goal is to basically somehow defer the debug attach step for at least a moment so things can fire up properly.

I've added "noDebug":true to my launch configuration and am using Chrome to debug instead, for now. Is that really all I can do? Would passing some sort of extra "delay" parameter to launch config be a big deal to implement?

**Comment by roblourens** [comment]
I don't want to add it if it's just a workaround for a bug in Node.

**Comment by connor4312** [comment]
The associated issues on node are closed and I'm unable to repro in 10.16. I tried a synchronous `import x from './module'` as well as the async `import('./module')` statement, and also tried inserting:

```
const target = Date.now() + 2000;
while (Date.now() < target) {}
```

to make sure the debugger would attach before hitting these statements. Everything seemed to work well.

**Comment by roblourens** [comment]
Thanks!
